### DISCUSSION TITLE
SO install. salt error

### DISCUSSION THREAD

**Discussion by osotob** [Security-Onion-Solutions/securityonion#6435] [Unsupported Versions] [answered]
Hi
I'm trying to install SO using iso or network, version 2.3.80 or 2.3.90. Always fails the installation:

2021-12-02T15:40:00Z | I | Putting system in state to run setup again
[ERROR   ] DNS lookup or connection check of 'salt' failed.
[ERROR   ] Master hostname: 'salt' not found or not responsive. Retrying in 30 seconds

Any help please?

**Comment by m0duspwnens** [comment, marked as answer]
Can you post the full sosetup.log? It appears that this is a reinstall. Did the first install fail as well or did you cancel it? If you cancelled it, do you recall at what point that was done and why you cancelled? Which type of node are you attempting to install?

**Reply by osotob** [reply]
-----------------------------
 Initializing Setup
-----------------------------

2021-12-02T15:50:18Z | I | Installing as the adminsonda user

-----------------------------
 System Characteristics
-----------------------------

2021-12-02T15:50:18Z | I | Executing command: uptime
 15:50:18 up 48 min,  2 users,  load average: 0.03, 0.03, 0.05
2021-12-02T15:50:18Z | I | Executing command: uname -a
Linux sondaso 3.10.0-1160.42.2.el7.x86_64 #1 SMP Tue Sep 7 14:49:57 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
2021-12-02T15:50:18Z | I | Executing command: free -h
              total        used        free      shared  buff/cache   available
Mem:           125G        1.8G        123G          9M        436M        122G
Swap:          8.0G          0B        8.0G
2021-12-02T15:50:18Z | I | Executing command: lscpu
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                20
On-line CPU(s) list:   0-19
Thread(s) per core:    2
Core(s) per socket:    10
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 85
Model name:            Intel(R) Xeon(R) Silver 4114 CPU @ 2.20GHz
Stepping:              4
CPU MHz:               2200.000
BogoMIPS:              4400.00
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              1024K
L3 cache:              14080K
NUMA node0 CPU(s):     0-19
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch epb cat_l3 cdp_l3 invpcid_single intel_ppin intel_pt ssbd mba ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp_epp pku ospke md_clear spec_ctrl intel_stibp flush_l1d
2021-12-02T15:50:18Z | I | Executing command: df -h
Filesystem               Size  Used Avail Use% Mounted on
devtmpfs                  63G     0   63G   0% /dev
tmpfs                     63G     0   63G   0% /dev/shm
tmpfs                     63G   10M   63G   1% /run
tmpfs                     63G     0   63G   0% /sys/fs/cgroup
/dev/mapper/system-root  293G  2.5G  291G   1% /
/dev/sda2                494M  141M  353M  29% /boot
/dev/sda1               1022M   12M 1011M   2% /boot/efi
/dev/mapper/system-tmp   2.0G   33M  2.0G   2% /tmp
/dev/mapper/system-nsm    11T  7.3G   11T   1% /nsm
tmpfs                     13G     0   13G   0% /run/user/1000
2021-12-02T15:50:18Z | I | Executing command: ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: eno3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 08:94:ef:71:12:4e brd ff:ff:ff:ff:ff:ff
3: eno4: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc mq state DOWN group default qlen 1000
    link/ether 08:94:ef:71:12:4f brd ff:ff:ff:ff:ff:ff
4: eno1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 08:94:ef:71:12:50 brd ff:ff:ff:ff:ff:ff
    inet x.x.x.x/28 brd x.x.x.x scope global noprefixroute eno1
       valid_lft forever preferred_lft forever
5: eno2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 08:94:ef:71:12:51 brd ff:ff:ff:ff:ff:ff
6: enp0s20f0u1u6: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UNKNOWN group default qlen 1000
    link/ether 0a:94:ef:71:07:61 brd ff:ff:ff:ff:ff:ff
7: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default
    link/ether 02:42:f4:8d:31:ec brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.1/16 brd 172.17.255.255 scope global docker0
       valid_lft forever preferred_lft forever
Detecting Base OS
Found OS: centos 7
Beginning Security Onion network install
Setup is running on TTY /dev/pts/0
2021-12-02T15:50:34Z | I | Disabling ipv6
net.ipv6.conf.all.disable_ipv6 = 1
net.ipv6.conf.default.disable_ipv6 = 1
Restarting Docker...
Detecting Base OS
Found OS: centos 7
Beginning pre-flight checks.
  Checking OS default repos with yum update.
Loaded plugins: versionlock
Not using downloaded base/repomd.xml because it is older than what we have:
  Current   : Thu Dec  2 11:02:50 2021
  Downloaded: Thu Dec  2 07:02:49 2021
Not using downloaded docker-ce-stable/repomd.xml because it is older than what we have:
  Current   : Thu Dec  2 11:13:42 2021
  Downloaded: Thu Dec  2 07:14:00 2021
Not using downloaded extras/repomd.xml because it is older than what we have:
  Current   : Thu Dec  2 11:08:28 2021
  Downloaded: Thu Dec  2 07:08:34 2021
Not using downloaded updates/repomd.xml because it is older than what we have:
  Current   : Thu Dec  2 11:12:27 2021
  Downloaded: Thu Dec  2 07:12:44 2021
Not using downloaded wazuh4_repo/repomd.xml because it is older than what we have:
  Current   : Thu Dec  2 11:15:29 2021
  Downloaded: Thu Dec  2 07:15:55 2021
Not using downloaded wazuh_repo/repomd.xml because it is older than what we have:
  Current   : Thu Dec  2 11:14:37 2021
  Downloaded: Thu Dec  2 07:14:59 2021

bash.x86_64                         4.2.46-35.el7_9             updates
bind-export-libs.x86_64             32:9.11.4-26.P2.el7_9.8     updates
bind-libs.x86_64                    32:9.11.4-26.P2.el7_9.8     updates
bind-libs-lite.x86_64               32:9.11.4-26.P2.el7_9.8     updates
bind-license.noarch                 32:9.11.4-26.P2.el7_9.8     updates
bind-utils.x86_64                   32:9.11.4-26.P2.el7_9.8     updates
binutils.x86_64                     2.27-44.base.el7_9.1        updates
containerd.io.x86_64                1.4.12-3.1.el7              docker-ce-stable
docker-ce.x86_64                    3:20.10.11-3.el7            docker-ce-stable
docker-ce-cli.x86_64                1:20.10.11-3.el7            docker-ce-stable
docker-ce-rootless-extras.x86_64    20.10.11-3.el7              docker-ce-stable
glibc.x86_64                        2.17-325.el7_9              updates
glibc-common.x86_64                 2.17-325.el7_9              updates
grub2.x86_64                        1:2.02-0.87.el7.centos.7    updates
grub2-common.noarch                 1:2.02-0.87.el7.centos.7    updates
grub2-efi-x64.x86_64                1:2.02-0.87.el7.centos.7    updates
grub2-pc.x86_64                     1:2.02-0.87.el7.centos.7    updates
grub2-pc-modules.noarch             1:2.02-0.87.el7.centos.7    updates
grub2-tools.x86_64                  1:2.02-0.87.el7.centos.7    updates
grub2-tools-extra.x86_64            1:2.02-0.87.el7.centos.7    updates
grub2-tools-minimal.x86_64          1:2.02-0.87.el7.centos.7    updates
httpd-tools.x86_64                  2.4.6-97.el7.centos.2       updates
kbd.x86_64                          1.15.5-16.el7_9             updates
kbd-legacy.noarch                   1.15.5-16.el7_9             updates
kbd-misc.noarch                     1.15.5-16.el7_9             updates
kernel.x86_64                       3.10.0-1160.49.1.el7        updates
kernel-tools.x86_64                 3.10.0-1160.49.1.el7        updates
kernel-tools-libs.x86_64            3.10.0-1160.49.1.el7        updates
libxml2.x86_64                      2.9.1-6.el7_9.6             updates
libxml2-python.x86_64               2.9.1-6.el7_9.6             updates
mdadm.x86_64                        4.1-8.el7_9                 updates
nspr.x86_64                         4.32.0-1.el7_9              updates
nss.x86_64                          3.67.0-3.el7_9              updates
nss-softokn.x86_64                  3.67.0-3.el7_9              updates
nss-softokn-freebl.x86_64           3.67.0-3.el7_9              updates
nss-sysinit.x86_64                  3.67.0-3.el7_9              updates
nss-tools.x86_64                    3.67.0-3.el7_9              updates
nss-util.x86_64                     3.67.0-1.el7_9              updates
openssh.x86_64                      7.4p1-22.el7_9              updates
openssh-clients.x86_64              7.4p1-22.el7_9              updates
openssh-server.x86_64               7.4p1-22.el7_9              updates
openssl.x86_64                      1:1.0.2k-22.el7_9           updates
openssl-devel.x86_64                1:1.0.2k-22.el7_9           updates
openssl-libs.x86_64                 1:1.0.2k-22.el7_9           updates
python-perf.x86_64                  3.10.0-1160.49.1.el7        updates
python36-distro.noarch              1.5.0-1.el7                 epel
rpm.x86_64                          4.11.3-48.el7_9             updates
rpm-build-libs.x86_64               4.11.3-48.el7_9             updates
rpm-libs.x86_64                     4.11.3-48.el7_9             updates
rpm-python.x86_64                   4.11.3-48.el7_9             updates
sudo.x86_64                         1.8.23-10.el7_9.2           updates
tzdata.noarch                       2021e-1.el7                 updates
wazuh-agent.x86_64                  4.2.5-1                     wazuh4_repo
  SUCCESS
  Checking repo URLs added by setup.
[INFO   ] Successfully reached https://download.docker.com/linux/centos/docker-ce.repo
[INFO   ] Successfully reached https://repo.securityonion.net/file/securityonion-repo/keys/SALTSTACK-GPG-KEY.pub
[INFO   ] Successfully reached https://download.docker.com/linux/ubuntu/gpg
[INFO   ] Successfully reached https://packages.wazuh.com/key/GPG-KEY-WAZUH
[INFO   ] Successfully reached https://packages.wazuh.com/3.x/yum/

  SUCCESS
  Checking various other URLs used by setup.
[INFO   ] Successfully reached https://raw.githubusercontent.com/Security-Onion-Solutions/securityonion/master/KEYS
[INFO   ] Successfully reached https://github.com/Neo23x0/signature-base
[INFO   ] Successfully reached https://sigs.securityonion.net/2.3.90/securityonion-2.3.90.iso.sig
[INFO   ] Successfully reached https://ghcr.io/
[INFO   ] Successfully reached https://rules.emergingthreats.net/open/
[INFO   ] Successfully reached https://rules.emergingthreatspro.com/

  SUCCESS

Pre-flight checks completed successfully!
2021-12-02T15:51:29Z | I | Executing command: systemctl enable NetworkManager
2021-12-02T15:51:29Z | I | Executing command: systemctl start NetworkManager
Testing if setup is running on a cloud instance...
MINION_ID = sondaso_standalone
Verifying all network devices are managed by Network Manager that should be
2021-12-02T15:52:10Z | I | Disabling Strelka rules: STRELKA='0'
Email is acceptable
Password is acceptable
Install summary:
Security Onion Version: 2.3.90
Node Type: STANDALONE
Hostname: sondaso
Management NIC: eno1
Management IP: x.x.x.x
Proxy:
  Server URL: https://x.x.x.x:8080
Bond NIC(s):
  - eno3
  - eno2
Home Network(s):
  - 10.0.0.0/8
  - 192.168.0.0/16
  - 172.16.0.0/12
Access URL: https://x.x.x.x
Allowed IP or Subnet: 0.0.0.0/0
Web User: sondaso@xxxx.es
Fleet User: sondaso@xxxx.es
Enabled Optional Components:
Metadata Tool: ZEEK
IDS Ruleset: ETOPEN
Patch Schedule:
  Type: auto
OS Package Updates: Open
Zeek Processes: 4
Suricata Processes: 4
Elasticsearch Heap Size: 25000m
Logstash Heap Size: 1000m
Logstash Worker Count: 20
Logstash Batch Size: 125
Logstash Input Threads: 1
Elasticsearch Storage Space: 5435GB
2021-12-02T15:53:04Z | I | Putting system in state to run setup again
[ERROR   ] DNS lookup or connection check of 'salt' failed.
[ERROR   ] Master hostname: 'salt' not found or not responsive. Retrying in 30 seconds
[ERROR   ] DNS lookup or connection check of 'salt' failed.
[ERROR   ] Master hostname: 'salt' not found or not responsive. Retrying in 30 seconds
[ERROR   ] DNS lookup or connection check of 'salt' failed.
[ERROR   ] Master hostname: 'salt' not found or not responsive. Retrying in 30 seconds
[ERROR   ] DNS lookup or connection check of 'salt' failed.
[ERROR   ] Master hostname: 'salt' not found or not responsive. Retrying in 30 seconds


-------------------------------


I tried to reinstall 3.80 and failed. I did a fresh install and i canceled at the proxy config point. The problem is that I cannot connect to it because i need to install a certificate. So the install program continued and after that failed to install. I installed the cert by ssh and tried the iso install afterwards. 

Thanks

**Reply by m0duspwnens** [reply]
@osotob I think I have identified the issue and fix for this. The root cause is the salt-minion package getting installed and so-setup stopping prior to configuring the salt-minion config file. This leaves the `master:` setting as `salt` since it is the default salt master for a newly installed salt-minion package. The fix is for the `salt-call` commands below to run locally as if no Salt master were present.

 We will be fixing this in the code, but if you want to proceed, below are the edits to make.

In `SecurityOnion/setup/so-functions` we will modify the following:
For version 2.3.80, these are line numbers 2004 and 2007. For 2.3.90, they are line numbers 2019 and 2022. You should see some lines that looks as follows:
```
			# Disable schedule so highstate doesn't start running during the install
			salt-call -l info schedule.disable

			# Kill any currently running salt jobs, also to prevent issues with highstate.
			salt-call -l info saltutil.kill_all_jobs
```

We will want to change append `--local` to the salt-call commands so that they look as follows:
```
			# Disable schedule so highstate doesn't start running during the install
			salt-call -l info schedule.disable --local

			# Kill any currently running salt jobs, also to prevent issues with highstate.
			salt-call -l info saltutil.kill_all_jobs --local
```

Thanks.

**Reply by osotob** [reply]
Hi m0duspwnens
Works like a charm!!

Thanks a lot!
